### PR TITLE
reset node_modules when current node version is different from the last version used

### PR DESCRIPTION
**Pull request by Jae**
it skips it if the last version is unknown, and just assume the current version is the last version used.

### CODE REVIEW

File: commands/install.js [hunk 1]
```diff
@@ -31,41 +31,63 @@ function getPackageManager(bosco, repoPath, interpreter) {
   return {name: name, command: nvm + command};
 }
 
+function cleanModulesIfVersionChanged(bosco, repoPath, repo, next) {
+  NodeRunner.getVersion(bosco, {cwd: repoPath}, function(err, currentVersion) {
+    var lastVersion = bosco.config.get('teams:' + bosco.getTeam() + ':nodes:' + repo);
+    if (lastVersion && lastVersion !== currentVersion) {
+      bosco.warn('Node version is changed from ' + lastVersion + ' to ' + currentVersion + ', clearing node_modules for ' + repoPath.blue);
+      exec('rm -rf ./node_modules', {cwd: repoPath}, function(err, stdout, stderr) {
```
Comment (gabceb) [inline commands/install.js]: I would NOT do this w/o first asking the user if its ok to wipe all the modules (AND also show the path you are about to delete). Just think about @bettiolo in an island sipping a pina colada and trying a new Node version (like he does) with crappy internet having to re-download the 600 mb node_modules folder of the payment gateway. He would not be a happy camper after this

Comment (gabceb) [inline commands/install.js]: IMO a better solution is to mimic what we do when we find a .nvmrc with a Node version that is not installed. Let the user know about it and only download the version if they pass a specific parameter on the command

**Comment by cliftonc** [comment]
Thinking about this more, we should definitely prompt to ask if they want to wipe the folder, but don't require a separate flag to trigger the behaviour - in most instances this is what people will want to happen (unlike some of the other things we have put behind flags).

If you switch node versions in .nvmrc it's highly likely the app won't work at all.

File: commands/install.js (hunk 1 above)
Comment (Jae) [inline commands/install.js]: ah... yes I haven't thought about Pina Colada effect....

Comment (codeaholics) [inline commands/install.js]: Never forget the Piña Colada effect!

Comment (bettiolo) [inline commands/install.js]: npm rebuild instead?

**Comment by bettiolo** [comment]
@cliftonc node 8 is moving towards NAPI which should not require native module rebuilds anymore:
https://github.com/nodejs/abi-stable-node

**Comment by cliftonc** [comment]
lgtm

**Comment by bettiolo** [comment]
Why didn't we do a npm rebuilt? We should also do it only with node versions that don't have the standard NPAPI.

**Comment by geophree** [comment]
An `npm rebuild` is a fine solution if you don't have connectivity, but if you've changed npm versions, we should really re-install the whole tree, because it can be significantly different.

**Pull request merged**